player/lua/console.lua

```diff
@@ -79,16 +79,10 @@ update_timer = mp.add_periodic_timer(0.05, function()
 end)
 update_timer:kill()
 
-utils.shared_script_property_observe("osc-margins", function(_, val)
+utils.user_data_observe("osc/margins", "native", function(_, val)
     if val then
-        -- formatted as "%f,%f,%f,%f" with left, right, top, bottom, each
-        -- value being the border size as ratio of the window size (0.0-1.0)
-        local vals = {}
-        for v in string.gmatch(val, "[^,]+") do
-            vals[#vals + 1] = tonumber(v)
-        end
-        global_margin_top = vals[3] -- top
-        global_margin_bottom = vals[4] -- bottom
+        global_margin_top = val.t -- top
+        global_margin_bottom = val.b -- bottom
```

Comment:
We could store the table directly in a `global_margin` (or just `margins` and then change any use of them to `global_margin.t` instead of de-structuring it.

Reply:
Done.